cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperatorMockTest.java

```diff
@@ -589,68 +419,64 @@ public void testReconcileUpdatesKafkaStorageType(Params params, VertxTestContext
+                kafkaAssembly(NAMESPACE, CLUSTER_NAME).patch(updatedStorageKafka);
 
                 LOGGER.info("Updating with changed storage type");
             })))
             .compose(v -> operator.reconcile(new Reconciliation("test-trigger", Kafka.RESOURCE_KIND, NAMESPACE, CLUSTER_NAME)))
             .onComplete(context.succeeding(v -> context.verify(() -> {
                 // Check the Volumes and PVCs were not changed
-                assertPVCs(context, KafkaResources.kafkaStatefulSetName(CLUSTER_NAME), originalPVCs.get());
-                assertVolumes(context, KafkaResources.kafkaStatefulSetName(CLUSTER_NAME), originalVolumes.get());
-                assertInitContainers(context, KafkaResources.kafkaStatefulSetName(CLUSTER_NAME), originalInitContainers.get());
+                assertPVCs(context, CLUSTER_NAME + "-kafka");
+                assertVolumes(context, CLUSTER_NAME + "-kafka");
                 async.flag();
             })));
     }
 
-    private void assertPVCs(VertxTestContext context, String statefulSetName, List<PersistentVolumeClaim> originalPVCs) {
-        context.verify(() -> {
-            StatefulSet statefulSet = client.apps().statefulSets().inNamespace(NAMESPACE).withName(statefulSetName).get();
-            assertThat(statefulSet, is(notNullValue()));
-            List<PersistentVolumeClaim> pvcs = statefulSet.getSpec().getVolumeClaimTemplates();
-            assertThat(originalPVCs.size(), is(pvcs.size()));
-            assertThat(originalPVCs, is(pvcs));
-        });
 
-    }
-
-    private void assertVolumes(VertxTestContext context, String statefulSetName, List<Volume> originalVolumes) {
+    private void assertPVCs(VertxTestContext context, String podSetName) {
         context.verify(() -> {
-            StatefulSet statefulSet = client.apps().statefulSets().inNamespace(NAMESPACE).withName(statefulSetName).get();
-            assertThat(statefulSet, is(notNullValue()));
-            List<Volume> volumes = statefulSet.getSpec().getTemplate().getSpec().getVolumes();
-            assertThat(originalVolumes.size(), is(volumes.size()));
-            assertThat(originalVolumes, is(volumes));
+            StrimziPodSet sps = supplier.strimziPodSetOperator.client().inNamespace(NAMESPACE).withName(podSetName).get();
+            assertThat(sps, is(notNullValue()));
+            List<PersistentVolumeClaimVolumeSource> persistentVolumeClaim = new ArrayList<>();
+            // We need to loop through the pods to make sure they have the right values
+            sps.getSpec().getPods().stream().map(PodSetUtils::mapToPod).forEach(pod -> {
+                pod.getSpec().getVolumes().forEach(volume -> {
+                    if (volume.getPersistentVolumeClaim() != null) {
+                        persistentVolumeClaim.add(volume.getPersistentVolumeClaim());
+                    }
+                });
+            });
+            assertThat(persistentVolumeClaim.size(), is(3));
```

Comment:
These are not PVCs. These are volumes in the Pods / containers. You need to assert the actual PersistentVolumeClaim Kubernetes resources.